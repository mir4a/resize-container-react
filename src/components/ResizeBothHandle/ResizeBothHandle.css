.resize-both-handle {
  right: calc(var(--resize-handle-size) / -2);
  bottom: calc(var(--resize-handle-size) / -2);
}

.resize-both-handle.rtl {
  right: unset;
  left: calc(var(--resize-handle-size) / -2);
  bottom: calc(var(--resize-handle-size) / -2);
}

.resize-both-handle:hover {
  cursor: nwse-resize;
}

.resize-both-handle.rtl:hover {
  cursor: nesw-resize;
}
