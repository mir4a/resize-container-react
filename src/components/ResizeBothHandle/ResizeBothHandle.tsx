import * as React from "react";
import useResizeHandle, { ResizeType } from "../../hooks/use-resize-handle";
import { ReactComponent as ArrowNWSE } from "../icons/ArrowNWSE.svg";
import "./ResizeBothHandle.css";

export interface ResizeBothHandleProps {
  onDrag: ({ deltaX, deltaY }: { deltaX?: number; deltaY?: number }) => void;
  onDragEnd: () => void;
  children?: React.ReactNode;
}

const ResizeBothHandle: React.FC<ResizeBothHandleProps> = ({
  children,
  onDrag,
  onDragEnd,
}) => {
  const ref = useResizeHandle({
    onDrag,
    onDragEnd,
    resizeType: ResizeType.Both,
  });

  return (
    <div className="resize-handle resize-both-handle" ref={ref}>
      {children ? children : <ArrowNWSE />}
    </div>
  );
};

export default ResizeBothHandle;
