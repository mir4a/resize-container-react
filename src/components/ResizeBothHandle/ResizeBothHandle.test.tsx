import React from "react";
import { render } from "@testing-library/react";
import ResizeBothHandle from "./ResizeBothHandle";

describe("ResizeBothHandle", () => {
  it("matches snapshot", () => {
    const { container } = render(<ResizeBothHandle />);
    expect(container).toMatchSnapshot();
  });
});
