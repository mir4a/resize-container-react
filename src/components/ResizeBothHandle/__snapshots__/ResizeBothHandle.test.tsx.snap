// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeBothHandle matches snapshot 1`] = `
<div>
  <div
    class="resize-handle resize-both-handle"
  >
    <div />
  </div>
</div>
`;
